test/t8n/t8n.cpp

```diff
@@ -0,0 +1,174 @@
+            std::cerr << "---\nCOINBASE: " << std::setw(2) << hex(block.coinbase) << "\n";
+        }
+
+        auto state = pre_state;
+
+        std::vector<std::pair<std::string, state::Transaction>> hash_txs;
+
+        json::json j_result;
+        j_result["currentDifficulty"] = "0x20000"; /* + current_difficulty;*/
+
+        if (!txs_file.empty())
+        {
+            evmc::VM vm{evmc_create_evmone(), {{"O", "0"}}};
+
+            std::string txs;
+            std::ifstream txs_f{txs_file};
+            std::copy(std::istream_iterator<char>{txs_f}, std::istream_iterator<char>{},
+                std::back_inserter(txs));
+            std::cerr << "---\nTXS:\n" << txs << "\n";
+
+            const auto j_txs = json::json::parse(std::ifstream{txs_file});
+            std::vector<state::Log> txs_logs;
+
+            if (j_txs.is_array())
+            {
+                j_result["receipts"] = json::json::array();
+                j_result["rejected"] = json::json::array();
+                for (auto& j_tx : j_txs)
+                {
+                    state::Transaction tx = test::from_json<state::Transaction>(j_tx);
+                    auto const tx_logs = state::transition(state, block, tx, rev, vm);
+                    if (tx_logs.has_value())
+                    {
+                        txs_logs.insert(txs_logs.begin(), (*tx_logs).begin(), (*tx_logs).end());
+                        auto& j_receipt = j_result["receipts"][j_result["receipts"].size()];
+                        j_receipt["transactionHash"] = j_tx["hash"];
+                        j_receipt["blockHash"] =
+                            "0x0000000000000000000000000000000000000000000000000000000000000000";
+                        j_receipt["contractAddress"] = "0x0000000000000000000000000000000000000000";
+                        j_receipt["gasUsed"] =
+                            "0xa868";  // FIXME: Get it from transaction execution.
+                        j_receipt["cumulativeGasUsed"] = j_receipt["gasUsed"];
+                        j_receipt["logs"] = json::json::array();  // FIXME: Add to_json<state:Log>
+                        j_receipt["logsBloom"] = "0x" + std::string(512, '0');
+                        j_receipt["root"] = "";
+                        j_receipt["status"] = "0x1";
+                        j_receipt["transactionIndex"] = "0x0";
+                    }
+                    else
+                        j_result["rejected"].push_back(j_tx["hash"]);
+                }
+            }
+
+            j_result["logsHash"] = "0x" + hex(keccak256(rlp::encode(txs_logs)));
+            j_result["stateRoot"] = "0x" + hex(state::mpt_hash(state.get_accounts()));
+        }
+
+        j_result["logsBloom"] = "0x" + std::string(512, '0');
+        j_result["receiptsRoot"] =
+            "0x0000000000000000000000000000000000000000000000000000000000000000";
+        j_result["txRoot"] = "0x0000000000000000000000000000000000000000000000000000000000000000";
```

Comment:
Could we have an overloaded version of `hex` to support various types, and then use something like `hex(intx::uint256{0})`?

Reply:
Or perhaps have these constants pre-defined at the top of the file and reused multiple times (e.g. `NULL_HEXSTRING_160`, `NULL_HEXSTRING_256`, `NULL_HEXSTRING_512`)